src/sentry/tasks/integrations.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import absolute_import
+
+from sentry.tasks.base import instrumented_task, retry
+
+from sentry.models import ExternalIssue, GroupLink, Integration
+
+
+@instrumented_task(
+    name='sentry.tasks.integrations.post_comment',
+    queue='integrations',
+    default_retry_delay=60 * 5,
+    max_retries=5
+)
+# TODO(jess): Add retry exclusions once ApiClients have better error handling
+@retry()
+def post_comment(project_id, group_id, data, **kwargs):
+    # sync Sentry comments to external issues
+    external_issues = list(
+        ExternalIssue.objects.filter(
+            id__in=GroupLink.objects.filter(
+                project_id=project_id,
+                group_id=group_id,
+                linked_type=GroupLink.LinkedType.issue,
+            ).values_list('linked_id', flat=True)
+        )
+    )
+
+    if external_issues:
+        integrations = {
+            i.id: i for i in Integration.objects.filter(
+                id__in=[external_issue.integration_id for external_issue in external_issues]
+            )
+        }
+
+        for external_issue in external_issues:
+            integration = integrations[external_issue.integration_id]
+            integration.get_installation().create_comment(external_issue.key, data['text'])
```

Comment:
hmm good point. i'm moving some stuff from the old plugins that will allow me to actually handle errors. but i'm also wondering if we should separate these out so that it's one external issue per task? that seems like the best way to make sure we won't duplicate comments but also can retry on some request errors.

Reply:
I think the answer to this is complicated, and depends on lots of things I don't have the answer for.

So like, there are potentially transient errors, and potentially fatal errors. In a transient error, you're right, you'd want to retry. But in a fatal error, you wouldn't, because a retry isn't going to work. So an example of transient is, the service you're talking to is down. And a fatal is, credentials are wrong or some other configuration is wrong.

If you want to properly handle transient errors, then yeah, we should fan each of these into their own tasks. If we don't want to retry at all and just handle any failure safely, just wrap in `safe_execute` which is what we do in most other places.

The downside to this is we don't have a way to message back to say like, "posting this comment failed because of X".

Reply:
ok i think fanning out is prob the better way to go then. i'll update this to do that. thanks for the help!